engine/src/execution_graph/logic_controllers/BatchJoinProcessing.h

```diff
@@ -502,6 +461,19 @@ class JoinPartitionKernel : public kernel {
 		this->query_graph = query_graph;
 		this->input_.add_port("input_a", "input_b");
 		this->output_.add_port("output_a", "output_b");
+
+		// lets parse part of the expression here, because we need the joinType before we load
+		std::string new_join_statement;
```

Comment:
you can consider making JoinRelatedKernels inherit from the same kernel interface to avoid having this initialization step in multiple places.

Reply:
I do agree its super redundant, but I will put it into a utility function. I dont like the idea of making an interface just for one utility function

Reply:
its to keep it RAII like the class that constructs should do the initializing of thigns as a general principle

Reply:
Yeah, and things are getting initialized in the constructor